src/Externals/mkdocs.jl

```diff
@@ -1 +1,72 @@
+const INDENT_WIDTH = 4
+pad(indents) = (" "^INDENT_WIDTH)^indents
 
+# write a mkdocs yaml file: takes settings from ``rmd`` RenderedMarkdown
+function writemkdocs(rmd)
+    abs_path = joinpath(dirname(rmd.outdir), "mkdocs.yml")
+    pathdir = dirname(abs_path)
+    ispath(pathdir) || mkpath(pathdir)
+    open(abs_path, "w") do f
+        info("writing `MkDocs` yaml file to $(abs_path)")
+        writemkdocs!(f, rmd)
+    end
+end
+
+function writemkdocs!(io::IO, rmd)
+    keys = [:site_name, :repo_url, :site_description, :site_author, :docs_dir]
+    optionalkeys = [:theme, :theme_dir, :copyright, :google_analytics, :markdown_extensions]
+    conf = deepcopy(rmd.document.data)
+    conf[:docs_dir] = basename(rmd.outdir)
+    haskey(conf, :site_name) || (conf[:site_name] = conf[:title])
+    [haskey(conf, k) ||
+        throw(ArgumentError("Document '$(conf[:title])' has no config key ':$k'.")) for k in keys]
+
+    [haskey(conf, k) && push!(keys, k) for k in optionalkeys]
```

Comment:
These unassigned comprehensions confused me for a moment. Neat though! I'm slightly apprehensive about them though, could be clearer with plain old ``for``-loops?

The comprehensions are allocating arrays, though not a high priority for this bit of code I'd think. Something to keep in mind.